CONVENTIONS.md

```diff
@@ -0,0 +1,55 @@
+# Coding Conventions
+
+The following document outlines a series of conventions that must be used for all code contributions to Fabric API, and recommendations for other modding projects.
+
+## General
+
+All code should meet the Yarn naming conventions which can be found [here](https://github.com/FabricMC/yarn/blob/20w21a/CONVENTIONS.md).
+
+All calls of associated methods and uses of class fields should be prefixed with `this.` For example, setting the field `name` on a class should always be done by calling `this.name = "name"` rather than `name = "name"`. This is especially pertinent in constructors where constructor parameters often share a name with class fields.
+
+## Annotations
+
+All methods that can potentially return a null value should be annotated in-line using the JetBrains `@Nullable` right before the return type definition for the method. For example, a method named `possiblyGetName` that can potentially return a null value should be written as `public @Nullable String possiblyGetName()...`.
```

Comment:
Arguably this is a whole entire issue in itself here. Unsure if this should be handled in a future pull request